### ISSUE TITLE
Datasette should work with Python 3.8 (and drop compatibility with Python 3.5)

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#622]
See #595, #594, #404.

The big thing holding me back from ditching Python 3.5 was glitch.com - but they now offer Python 3.7: https://support.glitch.com/t/can-you-upgrade-python-to-latest-version/7980/25?u=simonw

**Comment by simonw** [comment]
Here's the error I'm getting with Python 3.8 at the moment:
```
~/Dropbox/Development/datasette $ venv-py3.8.0/bin/datasette --memory -p 8055
Serve! files=() (immutables=()) on port 8055
Traceback (most recent call last):
  File "venv-py3.8.0/bin/datasette", line 11, in <module>
    load_entry_point('datasette', 'console_scripts', 'datasette')()
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/simonw/Dropbox/Development/datasette/datasette/cli.py", line 365, in serve
    uvicorn.run(ds.app(), host=host, port=port, log_level="info")
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/uvicorn/main.py", line 279, in run
    server.run()
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/uvicorn/main.py", line 305, in run
    self.config.setup_event_loop()
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/uvicorn/config.py", line 218, in setup_event_loop
    loop_setup()
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/uvicorn/loops/auto.py", line 3, in auto_loop_setup
    import uvloop
  File "/Users/simonw/Dropbox/Development/datasette/venv-py3.8.0/lib/python3.8/site-packages/uvloop/__init__.py", line 7, in <module>
    from .loop import Loop as __BaseLoop  # NOQA
  File "uvloop/includes/stdlib.pxi", line 114, in init uvloop.loop
AttributeError: module 'sys' has no attribute 'set_coroutine_wrapper'
```
I'm pretty confident upgrading uvicorn (and uvloop with it) will fix this.

**Comment by simonw** [comment]
Confirmed: upgrading to `uvicorn~=0.10.4` fixed it.

**Comment by simonw** [comment]
Here's where Python 3.5 started failing: https://travis-ci.org/simonw/datasette/builds/610682852

**Comment by simonw** [comment]
Released in [Datasette 0.31](https://datasette.readthedocs.io/en/latest/changelog.html#v0-31).
